---
string hint0
string hint1
string hint2
string hint3

